src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRequestForInformationJoint.java

```diff
@@ -0,0 +1,119 @@
+import uk.gov.hmcts.divorce.caseworker.service.notification.RequestForInformationNotification;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.RequestForInformation;
+import uk.gov.hmcts.divorce.divorcecase.model.RequestForInformationList;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+import uk.gov.hmcts.divorce.notification.NotificationDispatcher;
+import uk.gov.hmcts.divorce.notification.exception.NotificationTemplateException;
+
+import java.util.Collections;
+import java.util.List;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingDocuments;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CITIZEN;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.JUDGE;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class CaseworkerRequestForInformationJoint implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CASEWORKER_REQUEST_FOR_INFORMATION_JOINT = "caseworker-request-for-information-joint";
+
+    public static final String REQUEST_FOR_INFORMATION_NOTIFICATION_FAILED_ERROR
+        = "Unable to send Request for Information Notification for Case Id: ";
+
+    private final CaseworkerRequestForInformationHelper helper;
+
+    private final RequestForInformationNotification requestForInformationNotification;
+
+    private final NotificationDispatcher notificationDispatcher;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REQUEST_FOR_INFORMATION_JOINT)
+            .forAllStates()
+            .name("Request For Information")
+            .description("Request for information")
+            .showCondition("applicationType=\"jointApplication\"")
+            .showSummary()
+            .showEventNotes()
+            .endButtonLabel("Submit")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grantHistoryOnly(SUPER_USER, LEGAL_ADVISOR, JUDGE, SOLICITOR, CITIZEN, JUDGE))
+            .page("requestForInformation", this::midEvent)
+            .pageLabel("Request For Information")
+            .complex(CaseData::getRequestForInformationList)
+                .complex(RequestForInformationList::getRequestForInformation)
+                    .mandatory(RequestForInformation::getRequestForInformationJointParties)
```

Comment:
We could set the options that will be available to choose from as the request for information parties in an aboutToStart or have 2 versions of the input with a show condition based on application type? It could help us avoid duplicating the event

Reply:
If you setvup the list in aboutToStart you have to use a dynamic list, which doesn't work with show conditions, so you can't hide/show the text inputs on demand unfortunately.  A fixed list is the only way to handle those.

If you use multiple dropdowns, you need a valid show condition.  They need to be based on a field used in the event, and the event doesn't require applicationType, which means you end up with a visible field that isn't necessary.  I spent a while trying to work around that, but couldn't find an obvious solution.

Reply:
For a show condition to work, I was thinking we could get application type by using a readOnly field that's always hidden - we have similar config in CaseworkerOfflineDocumentVerified:
`.readonlyNoSummary(CaseData::getApplicationType, ALWAYS_HIDE)`

Reply:
Resolved, per slack discussion.